hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/health/GatewayBaselineDao.java

```diff
@@ -57,47 +55,53 @@
             .setEndTimeMillis(entitiesRequest.getEndTimeMillis())
             .addAllEntityIds(
                 entitiesResponse.getEntityList().stream()
-                    .map(x -> x.getId())
+                    .map(Entity::getId)
                     .collect(Collectors.toList()))
             .setEntityType(entitiesRequest.getEntityType())
             .addAllBaselineAggregateRequest(
-                getFunctionExpression(entitiesRequest, baselineRequestMap))
+                getFunctionExpressionList(entitiesRequest, baselineRequestMap))
             .addAllBaselineMetricSeriesRequest(
                 getBaselineTimeSeriesRequest(entitiesRequest, baselineRequestMap))
             .build();
     return Single.just(baselineEntitiesRequest);
   }
 
-  private Map<String, Baseline> getBaselineRequestMap(EntityRequest request) {
-    Map<String, Baseline> baselineMap = new HashMap<>();
-    List<MetricRequest> metricRequests = request.metricRequests();
-    if (metricRequests.size() > 0) {
-      for (MetricRequest metricRequest : metricRequests) {
-        baselineMap.putAll(
-            metricRequest.aggregationRequests().stream()
-                .filter(x -> x.baseline() != null)
-                .collect(Collectors.toMap(x -> x.alias(), y -> y.baseline())));
-        baselineMap.putAll(
-            metricRequest.seriesRequests().stream()
-                .filter(x -> x.aggregationRequest().baseline() != null)
-                .collect(Collectors.toMap(x -> x.alias(), y -> y.aggregationRequest().baseline())));
-      }
-    }
+  private Map<String, Boolean> getBaselineRequestMap(EntityRequest request) {
+    Map<String, Boolean> baselineMap =
+        request.metricRequests().stream()
+            .flatMap(
+                metricRequest ->
+                    metricRequest.aggregationRequests().stream()
+                        .filter(MetricAggregationRequest::baseline))
+            .collect(
+                Collectors.toMap(
+                    MetricAggregationRequest::alias, MetricAggregationRequest::baseline));
+    baselineMap.putAll(
+        request.metricRequests().stream()
+            .flatMap(
+                metricRequest ->
+                    metricRequest.seriesRequests().stream()
+                        .filter(
+                            timeSeriesRequest -> timeSeriesRequest.aggregationRequest().baseline()))
+            .collect(
+                Collectors.toMap(
+                    MetricSeriesRequest::alias,
+                    seriesRequest -> seriesRequest.aggregationRequest().baseline())));
     return baselineMap;
   }
 
-  private List<FunctionExpression> getFunctionExpression(
-      EntitiesRequest entitiesRequest, Map<String, Baseline> requestMap) {
-    List<Expression> selectionlist = entitiesRequest.getSelectionList();
+  private List<FunctionExpression> getFunctionExpressionList(
+      EntitiesRequest entitiesRequest, Map<String, Boolean> requestMap) {
```

Comment:
This requestmap is only being used to detect presence - a Set would be a better fit.

Reply:
right, fixed it